### PR TITLE
[swss_vars] Add 'resource_type' attribute

### PR DESCRIPTION
**Pull request by nazariig**
Signed-off-by: Nazarii Hnydyn <nazariig@nvidia.com>

#### Why I did it
* To support new configuration attribute

#### How I did it
* N/A

#### How to verify it
* N/A

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [x] 202012

#### Description for the changelog

* N/A

#### A picture of a cute animal (not mandatory but encouraged)
```
      .---.        .-----------
     /     \  __  /    ------
    / /     \(  )/    -----
   //////   ' \/ `   ---
  //// / // :    : ---
 // /   /  /`    '--
//          //..\\
       ====UU====UU====
           '//||\\`
             ''``
```

### CODE REVIEW

**Review by prsunny — APPROVED** [review]

**Review by anish-n — APPROVED** [review]

**Comment by anish-n** [comment]
@nazariig would you want to add backport tag for 202012?

**Comment by nazariig** [comment]
> 
> 
> @nazariig would you want to add backport tag for 202012?

Done

**Comment by nazariig** [comment]
@lguohan please have a look

File: files/build_templates/swss_vars.j2 [hunk 1]
```diff
@@ -2,5 +2,6 @@
     "asic_type": "{{ asic_type }}",
     "asic_id": "{{ DEVICE_METADATA.localhost.asic_id }}",
     "mac": "{{ DEVICE_METADATA.localhost.mac }}",
-    "synchronous_mode": {% if DEVICE_METADATA.localhost.synchronous_mode == "disable" %}"disable"{% else %}"enable"{% endif %}
+    "synchronous_mode": {% if DEVICE_METADATA.localhost.synchronous_mode == "disable" %}"disable"{% else %}"enable"{% endif %},
+    "resource_type": "{{ DEVICE_METADATA.localhost.resource_type }}"
```
Comment (lguohan) [inline files/build_templates/swss_vars.j2:6]: do we know if resource_type is always there or not, will this throw exception if resource_type is not defined?

Comment (prsunny) [reply files/build_templates/swss_vars.j2:6]: In my understanding, it will come as `"resource_type": ""` if not defined. @nazariig can you confirm.

Comment (nazariig) [reply files/build_templates/swss_vars.j2:6]: @lguohan / @prsunny please have a look at this example:
* **no value:**
```bash
root@sonic:/home/admin# cat /etc/sonic/config_db.json | jq -r '.DEVICE_METADATA'
{
  "localhost": {
    "buffer_model": "traditional",
    "default_bgp_status": "up",
    "default_pfcwd_status": "disable",
    "hostname": "sonic",
    "hwsku": "ACS-MSN4600",
    "mac": "1c:34:da:1e:95:00",
    "platform": "x86_64-mlnx_msn4600-r0"
  }
}

root@sonic:/# cat /usr/share/sonic/templates/swss_vars.j2
{
    "asic_type": "{{ asic_type }}",
    "asic_id": "{{ DEVICE_METADATA.localhost.asic_id }}",
    "mac": "{{ DEVICE_METADATA.localhost.mac }}",
    "synchronous_mode": {% if DEVICE_METADATA.localhost.synchronous_mode == "disable" %}"disable"{% else %}"enable"{% endif %},
    "resource_type": "{{ DEVICE_METADATA.localhost.resource_type }}"
}

root@sonic:/# sonic-cfggen -d -y /etc/sonic/sonic_version.yml -t /usr/share/sonic/templates/swss_vars.j2
{
    "asic_type": "mellanox",
    "asic_id": "",
    "mac": "1c:34:da:1e:95:00",
    "synchronous_mode": "enable",
    "resource_type": ""
}
```

* **default value:**
```bash
root@sonic:/home/admin# redis-cli -n 4 HSET 'DEVICE_METADATA|localhost' resource_type default
(integer) 1

root@sonic:/# sonic-cfggen -d -y /etc/sonic/sonic_version.yml -t /usr/share/sonic/templates/swss_vars.j2
{
    "asic_type": "mellanox",
    "asic_id": "",
    "mac": "1c:34:da:1e:95:00",
    "synchronous_mode": "enable",
    "resource_type": "default"
}
```

**Comment by nazariig** [comment]
@lguohan / @prsunny please have a look

**Comment by lguohan** [comment]
why is this not needed on the master branch?

**Comment by nazariig** [comment]
> 
> 
> why is this not needed on the master branch?

The PR is targeted to 202012 per @anish-n request.
In case we need it for master also, we can either cherry-pick or raise a new one.

**Pull request merged**